src/ansys/tools/visualization_interface/backends/pyvista/pyvista.py

```diff
@@ -439,12 +439,15 @@ def show(
         picked_objects_list = []
         if isinstance(plottable_object, list):
             # Keep them ordered based on picking
-            for meshobject in self._picked_list:
+            for meshobject in self._picked_list.values():
                 for elem in plottable_object:
                     if hasattr(elem, "name") and elem.name == meshobject.name:
                         picked_objects_list.append(elem)
-        elif hasattr(plottable_object, "name") and plottable_object in self._picked_list:
+        elif hasattr(plottable_object, "name") and plottable_object.name in self._picked_list:
             picked_objects_list = [plottable_object]
+        else:
+            # Return the picked objects in the order they were picked
+            picked_objects_list = list(self._picked_list.values())
```

Comment:
Hi @ahernsean - that is incorrect. Using a ``set()`` was providing us a random error truly... take this example code:

```py
# Initialize an empty set
my_set = set()

# Let's introduce the following values, in this order: "b", "f", "a", "c", "e", "d"
my_ordered_list = ["b", "f", "a", "c", "e", "d"]

# Introduction of the values in the set
[my_set.add(x) for x in my_ordered_list]

# The set will just make sure that the values are
# unique, but it will not keep the order
print(my_set)
```

![image](https://github.com/user-attachments/assets/77d7cbbe-699c-42cb-91cf-f3c2e15e5112)

As you can see in the output, set objects do not preserve order. They preserve uniqueness of values. 

Let's do the same now with a dictionary. As keys we will take our ordered_list again, and as values, their numerical order as per the alphabet.

```py
# Initialize an empty dictionary
my_dict = dict()

# Let's introduce the following values, in this order: "b", "f", "a", "c", "e", "d"
my_ordered_list = ["b", "f", "a", "c", "e", "d"]
my_ordered_values = [2, 6, 1, 3, 5, 4]

# Add the values to the dictionary
for key, val in zip(my_ordered_list, my_ordered_values):
    my_dict[key] = val

# Print the dictionary keys
print(my_dict.keys())

# Print the dictionary values
print(my_dict.values())
```

![image](https://github.com/user-attachments/assets/3d14460f-77c9-44e1-b0fe-144b14c6642a)

As you can see in the output, dictionary values are always printed in the key order of insertion. That's why we can just request the values and know, for a fact, that the picking order is preserved.

Let's try now to "deselect" letter "a" and reselect it... let's see how the order changes:

```py
# Initialize an empty dictionary
my_dict = dict()

# Let's introduce the following values, in this order: "b", "f", "a", "c", "e", "d"
my_ordered_list = ["b", "f", "a", "c", "e", "d"]
my_ordered_values = [2, 6, 1, 3, 5, 4]

# Add the values to the dictionary
for key, val in zip(my_ordered_list, my_ordered_values):
    my_dict[key] = val

print("After adding the values:")

# Print the dictionary keys
print(my_dict.keys())

# Print the dictionary values
print(my_dict.values())

# Deselect "a" and reselect it
my_dict.pop("a")
my_dict["a"] = 1

print("After deselecting and reselecting 'a':")

# Print the dictionary keys
print(my_dict.keys())

# Print the dictionary values
print(my_dict.values())
```

![image](https://github.com/user-attachments/assets/6508b39a-6e2f-411f-870e-ffaa624c02c0)


As you can see, the "deselected" element and "reselected" is inserted back again in the proper order.

Hope this makes you understand better the way both coding structures work!

Reply:
Looks like this is new as of Python 3.7. Thanks for the lesson!

Reply:
Correct! Thanks for finding that email!! 😄 that made my day.

Reply:
You gotta love Guido

![](https://i.giphy.com/media/v1.Y2lkPTc5MGI3NjExaDRqYm5wNGZhem9yc2Z1eGtiazk1azI1bDR4dDBnYjQzMjRiZmY1diZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/jXIK2C2XBsAVi/giphy.gif)